uefi/file_test.go

```diff
@@ -0,0 +1,121 @@
+package uefi
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var (
+	// File headers
+	// Hardcoded checksums for testing :(
+	// I don't know how to do it better without rewriting or calling code under test.
+	emptyPadHeader = append(FFGUID[:],
+		[]byte{8, EmptyBodyChecksum, byte(fvFileTypePad), 0, FileHeaderMinLength, 0x00, 0x00, 0xF8}...) // Empty pad file header with no data
+	goodFreeFormHeader = append(FFGUID[:],
+		[]byte{202, EmptyBodyChecksum, byte(fvFileTypeFreeForm), 0, FileHeaderMinLength, 0x00, 0x00, 0xF8}...) // Empty freeform file header with no data
+)
+
+var (
+	// File examples
+	emptyFile        = make([]byte, 0) // Empty file
+	emptyPadFile     = emptyPadHeader  // Empty pad file with no data
+	badFreeFormFile  []byte            // File with bad checksum. Should construct fine, but not validate
+	goodFreeFormFile []byte            // Good file
+)
+
+func init() {
+	goodFreeFormFile = append(goodFreeFormHeader, linuxSec...)
+	goodFreeFormFile = append(goodFreeFormFile, smallSec...)
+	goodFreeFormFile = append(goodFreeFormFile, []byte{0, 0}...) // Alignment
+	goodFreeFormFile = append(goodFreeFormFile, tinySec...)
+	goodFreeFormFile[20] = byte(FileHeaderMinLength + len(tinySec) + 2 + len(linuxSec) + len(smallSec))
+
+	badFreeFormFile = make([]byte, len(goodFreeFormFile))
+	copy(badFreeFormFile, goodFreeFormFile)
+	badFreeFormFile[16] = 0 // Zero out checksum
+}
+
+func TestValidate(t *testing.T) {
+	var tests = []struct {
+		buf  []byte
+		msgs []string
+	}{
+		{emptyPadFile, nil},
+		{badFreeFormFile, []string{"file FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF header checksum failure! sum was 54"}},
+		{goodFreeFormFile, nil},
+	}
+	for _, test := range tests {
+		f, err := NewFile(test.buf)
```

Comment:
Use subtests: `t.Run(test.name, func (t *testing.T) {...})`

Reply:
I didn't know about this! Thanks for the suggestion!